applications/jonny5/src/j5_authz_req.erl

```diff
@@ -198,21 +198,36 @@ maybe_authorize(Request, Limits) ->
 maybe_authorize_exception(Request, Limits) ->
     CallDirection = j5_request:call_direction(Request),
     AuthType = kz_json:get_value(<<"Authorizing-Type">>, j5_request:ccvs(Request)),
-    case not is_authorizing_mobile(AuthType)
-        andalso j5_request:classification(Request)
-    of
-        'false' ->
+    ResourceType = kz_json:get_value(<<"Resource-Type">>, j5_request:ccvs(Request)),
+    Classification = j5_request:classification(Request),
+
+    case is_authorizing_mobile(AuthType) of
+        'true' ->
             lager:debug("allowing mobile call"),
             j5_per_minute:authorize(Request, Limits);
-        <<"emergency">> ->
-            lager:debug("allowing emergency call"),
-            j5_request:authorize(<<"limits_disabled">>, Request, Limits);
-        <<"tollfree_us">> when CallDirection =:= <<"outbound">> ->
-            lager:debug("allowing outbound tollfree call"),
-            j5_request:authorize(<<"limits_disabled">>, Request, Limits);
-        _Else -> maybe_hard_limit(Request, Limits)
+        'false' ->
+            lager:debug("The resource type is ~s",[ResourceType]),
+            case  maybe_authorize_onnet(ResourceType, Limits) of
```

Comment:
Generally speaking we try to avoid more than 2 levels of nesting, preferring function clauses to case clauses.

For instance, the clause checking `Classification` should probably become a `maybe_authorize_by_classificiation(Request, Limits, Classification)` and do the checking of `Classification` in the function clauses.